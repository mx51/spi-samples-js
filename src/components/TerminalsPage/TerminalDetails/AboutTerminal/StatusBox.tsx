import React from 'react';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import { SPI_PAIR_STATUS } from '../../../../definitions/constants/commonConfigs';
import { ReactComponent as ConnectedIcon } from '../../../../images/ConnectedIcon.svg';
import { ReactComponent as ReconnectingIcon } from '../../../../images/ReconnectingIcon.svg';
import { ReactComponent as UnpairedIcon } from '../../../../images/UnpairedIcon.svg';

export default function StatusBox(classes: Any, status: string): React.ReactElement {
  switch (status) {
    case SPI_PAIR_STATUS.PairedConnected:
      return (
        <Box display="flex" alignItems="center" className={classes.statusBox}>
          <Typography align="right">
            <ConnectedIcon className={classes.connectedIcon} />
          </Typography>
          <Box display="flex" flexDirection="column" marginLeft={2}>
            <Typography variant="h5" className={classes.statusTitle}>
              Connected
            </Typography>
            <Typography variant="inherit" className={classes.statusText}>
              Ready
            </Typography>
          </Box>
        </Box>
      );
    case SPI_PAIR_STATUS.PairedConnecting:
      return (
        <Box display="flex" alignItems="center" className={classes.statusBox}>
          <Typography align="right">
            <ReconnectingIcon className={classes.reconnectIcon} />
          </Typography>
          <Box display="flex" flexDirection="column" marginLeft={2}>
            <Typography variant="h5" className={classes.statusTitle}>
              Connecting
            </Typography>
            <Typography variant="inherit" className={classes.statusText}>
              Pairing
            </Typography>
          </Box>
        </Box>
      );
    default:
      return (
        <Box display="flex" alignItems="center" className={classes.statusBox}>
          <Typography align="right">
            <UnpairedIcon className={classes.unpairedIcon} />
          </Typography>
          <Box display="flex" flexDirection="column" marginLeft={2}>
            <Typography variant="h5" className={classes.statusTitle}>
              Disconnected
            </Typography>
            <Typography variant="inherit" className={classes.statusText}>
              Unpaired
            </Typography>
          </Box>
        </Box>
      );
  }
}
