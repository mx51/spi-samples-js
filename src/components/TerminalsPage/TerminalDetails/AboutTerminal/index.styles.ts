import { createStyles, makeStyles, Theme } from '@material-ui/core';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    text: {
      fontWeight: 300,
    },
    detailRow: {
      margin: theme.spacing(3, 0, 0),
    },
    sectionSpacing: {
      width: '100%',
      marginTop: theme.spacing(3),
      paddingTop: theme.spacing(3),
      borderTop: `1px solid ${theme.palette.secondary.main}`,
    },
    fullWidth: {
      width: '100%',
    },
    actionButton: {
      display: 'block',
      color: theme.palette.info.main,
      background: 'transparent',
      border: 'none',
      cursor: 'pointer',
    },
    statusBox: {
      padding: theme.spacing(1.5, 1),
      marginBottom: theme.spacing(1.5),
      '& h5': {
        fontSize: '1.125rem',
        fontWeight: 500,
        wordBreak: 'break-word',
      },
      '& span': {
        fontSize: '0.875rem',
        fontWeight: 400,
        color: theme.palette.text.secondary,
        wordBreak: 'break-word',
      },
      '&>.MuiTypography-root': {
        display: 'flex',
        alignItems: 'center',
      },
    },
    unpairedIcon: {
      color: theme.palette.error.main,
    },
    reconnectIcon: {
      color: theme.palette.warning.main,
    },
    connectedIcon: {
      color: theme.palette.success.main,
    },
  })
);

export default useStyles;
