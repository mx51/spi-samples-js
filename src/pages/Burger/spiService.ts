import { Spi as SpiClient, SpiFlow, TerminalConfigurationResponse, TransactionOptions } from '@mx51/spi-client-js';

class Spi {
  _spi: any;
  _posId: string;
  _eftposAddress: string;
  _spiSecrets: any;
  _options: any;
  _posName: any;
  _version: any;
  _rcptFromEftpos: Boolean;
  _sigFlowFromEftpos: Boolean;
  _apiKey: any;
  _serialNumber: string;
  _tenantCode: string;
  _autoResolveEftposAddress: Boolean;
  _testMode: Boolean;
  _useSecureWebSockets: Boolean;
  _log: any;
  _printMerchantCopy: Boolean;

  constructor() {
    this._spi = null;
    this._posId = window.localStorage.getItem('posID') || '';
    this._eftposAddress = window.localStorage.getItem('eftpos_address') || '192.168.1.1';
    this._spiSecrets = null;
    this._options = null;
    this._version = '2.9.0';
    this._rcptFromEftpos = window.localStorage.getItem('rcpt_from_eftpos') === 'true';
    this._sigFlowFromEftpos = window.localStorage.getItem('check-sig-eftpos') === 'true';
    this._printMerchantCopy = window.localStorage.getItem('print_merchant_copy_input') === 'true';
    this._apiKey = window.localStorage.getItem('api_key') || '';
    this._serialNumber = window.localStorage.getItem('serial') || '';
    this._tenantCode = window.localStorage.getItem('tenant_code') || '';
    this._posName = 'mx51';
    this._autoResolveEftposAddress = window.localStorage.getItem('auto_address') === 'true';
    this._testMode = window.localStorage.getItem('test_mode') === 'true';
    this._useSecureWebSockets = window.localStorage.getItem('use_secure_web_sockets') === 'true';
    this._log = console;
    const secretsString = window.localStorage.getItem('secrets') || '';
    try {
      const secrets = JSON.parse(secretsString);
      if (secrets) {
        this._spiSecrets = secrets;
      }
    } catch (error) {
      this._log.error('unable to access secrets');
    }

    // Where there is an existing pairing and no tenant code set, default to wbc
    if (!this._tenantCode && this._spiSecrets) {
      this._tenantCode = 'wbc';
      window.localStorage.setItem('tenant_code', 'wbc');
    }

    this.getTenantsList();
  }

  start() {
    try {
      this._spi = new SpiClient(this._posId, this._serialNumber, this._eftposAddress, this._spiSecrets); // It is ok to not have the secrets yet to start with.
      this._spi.Config.PromptForCustomerCopyOnEftpos = this._rcptFromEftpos;
      this._spi.Config.SignatureFlowOnEftpos = this._sigFlowFromEftpos;
      this._spi.Config.PrintMerchantCopy = this._printMerchantCopy;
      this._spi.SetPosInfo(this._posName, this._version);
      this._spi.SetTenantCode(this._tenantCode);
      this._spi.SetDeviceApiKey(this._apiKey);
      this._options = new TransactionOptions();
      this._options.SetCustomerReceiptHeader(window.localStorage.getItem('receipt_header_input') || '');
      this._options.SetCustomerReceiptFooter(window.localStorage.getItem('receipt_footer_input') || '');
      this._options.SetMerchantReceiptHeader(window.localStorage.getItem('receipt_header_input') || '');
      this._options.SetMerchantReceiptFooter(window.localStorage.getItem('receipt_footer_input') || '');
    } catch (e) {
      this._log.info(e.Message);
      return;
    }
    this.onSpiStateChange = this.onSpiStateChange.bind(this);
    document.addEventListener('DeviceAddressChanged', this.onSpiStateChange);
    document.addEventListener('StatusChanged', this.onSpiStateChange);
    document.addEventListener('PairingFlowStateChanged', this.onSpiStateChange);
    document.addEventListener('SecretsChanged', Spi.onSecretsChange);
    document.addEventListener('TxFlowStateChanged', this.onSpiStateChange);
    this._spi.PrintingResponse = this.onSpiResponse.bind(this);
    this._spi.TerminalConfigurationResponse = Spi.onTerminalConfigurationChange;
    this._spi.TerminalStatusResponse = this.onSpiResponse.bind(this);
    this._spi.BatteryLevelChanged = this.onSpiResponse.bind(this);
    this._spi.TransactionUpdateMessage = Spi.onSpiTransactionUpdate.bind(this);
    this.setAutoAddressResolutionState();
    this._spi.Start();
    this.printStatusAndActions();
  }

  async getTenantsList() {
    const tenants = await SpiClient.GetAvailableTenants(this._posName, this._apiKey, 'AU');
    const defaultTenantList = [
      {
        code: 'wbc',
        name: 'Westpac Presto',
      },
      {
        code: 'till',
        name: 'Till Payments',
      },
    ];
    localStorage.setItem('tenants', JSON.stringify(tenants.Data.length ? tenants.Data : defaultTenantList));
  }

  onSpiStateChange(e: any) {
    if (e.detail && e.detail === 'PairedConnected') {
      this._spi.AckFlowEndedAndBackToIdle();
    }
  }

  static onSecretsChange(e: any) {
    window.localStorage.setItem('secrets', JSON.stringify(e.detail));
  }

  static onTerminalConfigurationChange(m: Message) {
    const terminalConfigurationResponse = new TerminalConfigurationResponse(m);
    window.localStorage.setItem('serial', terminalConfigurationResponse.GetSerialNumber());
  }

  static onSpiTransactionUpdate(m: Message) {
    document.dispatchEvent(new CustomEvent('TxnUpdateMessage', { detail: m }));
  }

  onSpiResponse() {
    this._spi.AckFlowEndedAndBackToIdle();
    this.printStatusAndActions();
  }

  setAutoAddressResolutionState() {
    this._spi.SetTestMode(this._testMode);
    this._spi.SetSecureWebSockets(this._useSecureWebSockets);
    this._spi.SetAutoAddressResolution(this._autoResolveEftposAddress);
  }

  printStatusAndActions() {
    this.printFlowInfo();
  }

  printFlowInfo() {
    switch (this._spi.CurrentFlow) {
      case SpiFlow.Pairing:
        this._log.info(this._spi.CurrentPairingFlowState);
        break;
      case SpiFlow.Transaction:
        this._log.info(this._spi.CurrentTxFlowState);
        break;
      default:
        // eslint-disable-next-line no-console
        console.log('Unable to handle flow state');
    }
  }
}

export { Spi as default };
