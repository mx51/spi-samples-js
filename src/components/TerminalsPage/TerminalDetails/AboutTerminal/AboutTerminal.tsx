import React from 'react';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { Link } from 'react-router-dom';
import { SpiStatus } from '@mx51/spi-client-js';
import { PATH_PAIR } from '../../../../definitions/constants/routerConfigs';
import {
  terminalConfigurationsPartOne,
  terminalConfigurationsPartTwo,
} from '../../../../definitions/constants/terminalConfigs';
import { useAppDispatch, useAppSelector } from '../../../../redux/hooks';
import { terminalPosRefId } from '../../../../redux/reducers/TerminalSlice/terminalsSliceSelectors';
import { handleUnPairClick } from '../../../../utils/common/pair/pairStatusHelpers';
import { settlement, settlementEnquiry } from '../../../../utils/common/terminal/terminalHelpers';
import { IAboutTerminal } from '../interfaces';
import useStyles from './index.styles';
import StatusBox from './StatusBox';
import { SPI_PAIR_STATUS } from '../../../../definitions/constants/commonConfigs';

export default function AboutTerminal({
  receiptToggle,
  setReceiptToggle,
  terminal,
}: IAboutTerminal): React.ReactElement {
  const dispatch = useAppDispatch();
  const posRefId = useAppSelector(terminalPosRefId(terminal.id));
  const classes = useStyles();

  const handleSettlementDisplay = () => {
    setReceiptToggle({
      settlement: !receiptToggle.settlement,
      settlementEnquiry: false, // ensure settlementEnquiry panel is closed when settlement panel is opened or closed
    });

    if (!receiptToggle.settlement) settlement(terminal.id, posRefId as string);
  };

  const handleSettlementEnquiryDisplay = () => {
    setReceiptToggle({
      settlement: false, // ensure settlement panel is closed when settlementEnquiry panel is opened or closed
      settlementEnquiry: !receiptToggle.settlementEnquiry,
    });

    if (!receiptToggle.settlementEnquiry) settlementEnquiry(terminal.id, posRefId as string);
  };

  return (
    <>
      <Grid container>
        <Grid item className={(classes.fullWidth, classes.sectionSpacing)}>
          <Typography variant="h6" component="h1">
            Terminal information
          </Typography>
          <Typography className={classes.text}>Information about this terminal</Typography>
          <Grid container className={classes.detailRow}>
            <Grid item md={4} xs={12}>
              <Typography>Pairing status</Typography>
            </Grid>
            <Grid item md={8} xs={12}>
              <StatusBox status={terminal?.status} />
              {terminal.status === SpiStatus.PairedConnected && (
                <>
                  <Button
                    className={classes.actionButton}
                    data-test-id="terminalDetailsSettlementBtn"
                    onClick={handleSettlementDisplay}
                  >
                    Settlement
                  </Button>
                  <Button
                    className={classes.actionButton}
                    data-test-id="terminalDetailsSettlementEnquiryBtn"
                    onClick={handleSettlementEnquiryDisplay}
                  >
                    Settlement enquiry
                  </Button>
                </>
              )}
              {[SpiStatus.PairedConnected, SpiStatus.PairedConnecting].includes(terminal.status) && (
                <Button
                  className={classes.actionButton}
                  data-test-id="terminalDetailsUnpairBtn"
                  onClick={() => handleUnPairClick(dispatch, terminal.id)}
                >
                  Unpair terminal
                </Button>
              )}
              {terminal.status === SpiStatus.Unpaired && (
                <Button
                  className={classes.actionButton}
                  component={Link}
                  data-test-id="terminalDetailsPairBtn"
                  to={`${PATH_PAIR}/${terminal.serialNumber}`}
                >
                  Pair terminal
                </Button>
              )}
            </Grid>
          </Grid>
          {terminalConfigurationsPartOne(terminal).map(({ title, content }) => (
            <Grid className={classes.detailRow} container key={title}>
              <Grid item md={4} xs={12}>
                <Typography>{title}</Typography>
              </Grid>
              <Grid item md={8} xs={12}>
                <Typography className={classes.text}>{content}</Typography>
              </Grid>
            </Grid>
          ))}
        </Grid>
        <Grid item className={(classes.fullWidth, classes.sectionSpacing)}>
          <Typography variant="h6" component="h1">
            Pairing configuration
          </Typography>
          {terminal?.status === SPI_PAIR_STATUS.Unpaired ? (
            ''
          ) : (
            <Typography className={classes.text}>
              To update the Pairing configuration you need to unpair this terminal using the Unpair button above
            </Typography>
          )}
          {terminalConfigurationsPartTwo(terminal).map(({ title, content }) => (
            <Grid className={classes.detailRow} container key={title}>
              <Grid item md={4} xs={12}>
                <Typography>{title}</Typography>
              </Grid>
              <Grid item md={8} xs={12}>
                <Typography className={classes.text}>{content}</Typography>
              </Grid>
            </Grid>
          ))}
        </Grid>
      </Grid>
    </>
  );
}
