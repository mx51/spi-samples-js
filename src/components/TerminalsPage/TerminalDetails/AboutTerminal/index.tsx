import React from 'react';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import {
  terminalConfigurationsPartOne,
  terminalConfigurationsPartTwo,
} from '../../../../definitions/constants/terminalConfigs';
import { useAppDispatch } from '../../../../redux/hooks';
import { handleUnPairClick } from '../../../../utils/common/pair/pairStatusHelpers';
import useStyles from './index.styles';
import StatusBox from './StatusBox';

export default function AboutTerminal({ receiptToggle, setReceiptToggle, terminal }: Any): React.ReactElement {
  const dispatch = useAppDispatch();
  const classes = useStyles();

  const handleReceiptPanelDisplay = () => {
    setReceiptToggle(!receiptToggle);
  };

  return (
    <>
      <Grid container>
        <Grid item className={(classes.fullWidth, classes.sectionSpacing)}>
          <Typography variant="h6" component="h1">
            Terminal information
          </Typography>
          <Typography className={classes.text}>Information about this terminal</Typography>
          <Grid container className={classes.detailRow}>
            <Grid item md={4} xs={12}>
              <Typography>Pairing status</Typography>
            </Grid>
            <Grid item md={8} xs={12}>
              {StatusBox(classes, terminal?.status)}
              <Button className={classes.actionButton} onClick={handleReceiptPanelDisplay}>
                Settlement
              </Button>
              <Button className={classes.actionButton} onClick={handleReceiptPanelDisplay}>
                Settlement enquiry
              </Button>
              <Button className={classes.actionButton} onClick={() => handleUnPairClick(dispatch, terminal.id)}>
                Unpair terminal
              </Button>
            </Grid>
          </Grid>
          {terminalConfigurationsPartOne(terminal).map(({ title, content }) => (
            <Grid className={classes.detailRow} container key={title}>
              <Grid item md={4} xs={12}>
                <Typography>{title}</Typography>
              </Grid>
              <Grid item md={8} xs={12}>
                <Typography className={classes.text}>{content}</Typography>
              </Grid>
            </Grid>
          ))}
        </Grid>
        <Grid item className={(classes.fullWidth, classes.sectionSpacing)}>
          <Typography variant="h6" component="h1">
            Pairing configuration
          </Typography>
          <Typography className={classes.text}>
            To update the Pairing configuration you need to unpair this terminal using the Unpair button above
          </Typography>
          {terminalConfigurationsPartTwo(terminal).map(({ title, content }) => (
            <Grid className={classes.detailRow} container key={title}>
              <Grid item md={4} xs={12}>
                <Typography>{title}</Typography>
              </Grid>
              <Grid item md={8} xs={12}>
                <Typography className={classes.text}>{content}</Typography>
              </Grid>
            </Grid>
          ))}
        </Grid>
      </Grid>
    </>
  );
}
