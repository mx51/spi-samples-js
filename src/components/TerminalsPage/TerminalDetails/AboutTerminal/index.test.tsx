import React from 'react';
import { cleanup, fireEvent } from '@testing-library/react';
import AboutTerminal from '.';
import {
  terminalConfigurationsPartOne,
  terminalConfigurationsPartTwo,
} from '../../../../definitions/constants/terminalConfigs';
import spiService from '../../../../services/spiService';
import mockWithRedux, {
  defaultMockTerminals,
  mockSpiClient,
  mockTerminalInstanceId,
} from '../../../../utils/tests/common';

describe('Test <AboutTerminal />', () => {
  let mockContainer: Any;

  beforeEach(() => {
    spiService.spiHardwarePrinterAvailable = jest.fn().mockReturnValue(false);
    mockContainer = mockWithRedux(
      <AboutTerminal
        receiptToggle={{
          settlement: false,
          settlementEnquiry: false,
        }}
        setReceiptToggle={jest.fn()}
        terminal={defaultMockTerminals[mockTerminalInstanceId]}
      />
    );
  });

  afterEach(cleanup);

  test('should match AboutTerminal snapshot test', () => {
    // Assert
    expect(mockContainer).toMatchSnapshot();
  });

  test('should show "%" with battery value when SPI response data is available', () => {
    // Arrange
    const terminal = {
      ...defaultMockTerminals[mockTerminalInstanceId],
      batteryLevel: '70',
    };

    // Act
    const terminalConfigResponse = terminalConfigurationsPartOne(terminal);

    // Assert
    expect(terminalConfigResponse[3].content).toBe('70%');
  });

  test('should show "Yes" value in terminal details page when address type is auto', () => {
    // Arrange
    const terminal = {
      ...defaultMockTerminals[mockTerminalInstanceId],
      autoAddress: true,
    };

    // Act
    const terminalConfigResponse = terminalConfigurationsPartTwo(terminal);

    // Assert
    expect(terminalConfigResponse[1].content).toBe('Yes');
  });

  test('should unpair button on click event function get defined', () => {
    // Arrange
    const handleUnPairClick = jest.fn();
    const terminalDetailsUnpairBtnDOM = mockContainer.querySelector('[data-test-id="terminalDetailsUnpairBtn"]');

    spiService.readTerminalInstance = jest.fn().mockReturnValue({
      spiClient: {
        ...mockSpiClient,
        Unpair: jest.fn(),
      },
    });

    // Act
    fireEvent.click(terminalDetailsUnpairBtnDOM);
    handleUnPairClick();

    // Assert
    expect(handleUnPairClick).toBeDefined();
    expect(spiService.readTerminalInstance(mockTerminalInstanceId).spiClient.Unpair).toBeCalled();
  });

  test('should toggle settlement panel when clicked settlement button', () => {
    // Arrange
    const handleSettlementDisplay = jest.fn();
    const terminalDetailsSettlementBtnDOM = mockContainer.querySelector(
      '[data-test-id="terminalDetailsSettlementBtn"]'
    );
    spiService.readTerminalInstance = jest.fn().mockReturnValue({
      spiClient: {
        ...mockSpiClient,
        InitiateSettleTx: jest.fn(),
      },
    });

    // Act
    fireEvent.click(terminalDetailsSettlementBtnDOM);
    handleSettlementDisplay();

    // Assert
    expect(handleSettlementDisplay).toBeDefined();
    expect(spiService.readTerminalInstance(mockTerminalInstanceId).spiClient.InitiateSettleTx).toBeCalled();
  });

  test('should toggle settlementEnquiry panel when clicked settlementEnquiry button', () => {
    // Arrange
    const handleSettlementEnquiryDisplay = jest.fn();
    const terminalDetailsSettlementEnquiryBtnBtnDOM = mockContainer.querySelector(
      '[data-test-id="terminalDetailsSettlementEnquiryBtn"]'
    );
    spiService.readTerminalInstance = jest.fn().mockReturnValue({
      spiClient: {
        ...mockSpiClient,
        InitiateSettlementEnquiry: jest.fn(),
      },
    });

    // Act
    fireEvent.click(terminalDetailsSettlementEnquiryBtnBtnDOM);
    handleSettlementEnquiryDisplay();

    // Assert
    expect(handleSettlementEnquiryDisplay).toBeDefined();
    expect(spiService.readTerminalInstance(mockTerminalInstanceId).spiClient.InitiateSettlementEnquiry).toBeCalled();
  });

  test('should call settlement() when settlement value is true', () => {
    // Arrange
    const settlement = jest.fn();
    const handleSettlementDisplay = jest.fn().mockImplementation(() => {
      settlement();
    });
    const newMockContainer = mockWithRedux(
      <AboutTerminal
        receiptToggle={{
          settlement: true,
          settlementEnquiry: false,
        }}
        setReceiptToggle={jest.fn()}
        terminal={defaultMockTerminals[mockTerminalInstanceId]}
      />
    );
    const terminalDetailsSettlementBtnDOM = newMockContainer.querySelector(
      '[data-test-id="terminalDetailsSettlementBtn"]'
    );
    spiService.readTerminalInstance = jest.fn().mockReturnValue({
      spiClient: {
        ...mockSpiClient,
        InitiateSettleTx: jest.fn(),
      },
    });

    // Act
    fireEvent.click(terminalDetailsSettlementBtnDOM);
    handleSettlementDisplay();

    // Assert
    expect(settlement).toBeCalled();
  });

  test('should call settlementEnquiry() when settlementEnquiry value is true', () => {
    // Arrange
    const settlementEnquiry = jest.fn();
    const handleSettlementEnquiryDisplay = jest.fn().mockImplementation(() => {
      settlementEnquiry();
    });
    const newMockContainer = mockWithRedux(
      <AboutTerminal
        receiptToggle={{
          settlement: false,
          settlementEnquiry: true,
        }}
        setReceiptToggle={jest.fn()}
        terminal={defaultMockTerminals[mockTerminalInstanceId]}
      />
    );
    const terminalDetailsSettlementEnquiryBtnBtnDOM = newMockContainer.querySelector(
      '[data-test-id="terminalDetailsSettlementEnquiryBtn"]'
    );
    spiService.readTerminalInstance = jest.fn().mockReturnValue({
      spiClient: {
        ...mockSpiClient,
        InitiateSettlementEnquiry: jest.fn(),
      },
    });

    // Act
    fireEvent.click(terminalDetailsSettlementEnquiryBtnBtnDOM);
    handleSettlementEnquiryDisplay();

    // Assert
    expect(settlementEnquiry).toBeCalled();
  });
});
