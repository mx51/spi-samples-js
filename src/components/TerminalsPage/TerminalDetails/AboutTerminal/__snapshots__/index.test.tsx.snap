// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AboutTerminal /> should match AboutTerminal snapshot test 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root makeStyles-sectionSpacing-3 MuiGrid-item"
    >
      <h1
        class="MuiTypography-root MuiTypography-h6"
      >
        Terminal information
      </h1>
      <p
        class="MuiTypography-root makeStyles-text-1 MuiTypography-body1"
      >
        Information about this terminal
      </p>
      <div
        class="MuiGrid-root makeStyles-detailRow-2 MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Pairing status
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
        >
          <div
            class="MuiBox-root MuiBox-root-12 makeStyles-statusBox-6"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight"
            >
              <svg
                class="makeStyles-unpairedIcon-9"
              >
                UnpairedIcon.svg
              </svg>
            </p>
            <div
              class="MuiBox-root MuiBox-root-13"
            >
              <h5
                class="MuiTypography-root makeStyles-statusTitle-7 MuiTypography-h5"
              >
                Disconnected
              </h5>
              <span
                class="MuiTypography-root makeStyles-statusText-8"
              >
                Unpaired
              </span>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-actionButton-5"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Settlement
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-actionButton-5"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Settlement enquiry
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-actionButton-5"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Unpair terminal
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-detailRow-2 MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Merchant ID
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
        >
          <p
            class="MuiTypography-root makeStyles-text-1 MuiTypography-body1"
          >
            undefined
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-detailRow-2 MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Terminal ID
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
        >
          <p
            class="MuiTypography-root makeStyles-text-1 MuiTypography-body1"
          >
            undefined
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-detailRow-2 MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Serial number
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
        >
          <p
            class="MuiTypography-root makeStyles-text-1 MuiTypography-body1"
          >
            undefined
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-detailRow-2 MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Battery
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
        >
          <p
            class="MuiTypography-root makeStyles-text-1 MuiTypography-body1"
          >
            undefined
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-sectionSpacing-3 MuiGrid-item"
    >
      <h1
        class="MuiTypography-root MuiTypography-h6"
      >
        Pairing configuration
      </h1>
      <p
        class="MuiTypography-root makeStyles-text-1 MuiTypography-body1"
      >
        To update the Pairing configuration you need to unpair this terminal using the Unpair button above
      </p>
      <div
        class="MuiGrid-root makeStyles-detailRow-2 MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Payment provider
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
        >
          <p
            class="MuiTypography-root makeStyles-text-1 MuiTypography-body1"
          >
            undefined
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-detailRow-2 MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Configuration option
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
        >
          <p
            class="MuiTypography-root makeStyles-text-1 MuiTypography-body1"
          >
            No
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-detailRow-2 MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            EFTPOS address
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
        >
          <p
            class="MuiTypography-root makeStyles-text-1 MuiTypography-body1"
          >
            undefined
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-detailRow-2 MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            POS ID
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
        >
          <p
            class="MuiTypography-root makeStyles-text-1 MuiTypography-body1"
          >
            undefined
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
